kong/tools/string.lua

```diff
@@ -151,6 +151,33 @@ function _M.bytes_to_str(bytes, unit, scale)
 end
 
 
+local SCALES = {
+  k = 1024,
+  K = 1024,
+  m = 1024 * 1024,
+  M = 1024 * 1024,
+  g = 1024 * 1024 * 1024,
+  G = 1024 * 1024 * 1024,
+}
+
+function _M.parse_ngx_size(str)
+  local len = #str
```

Comment:
let'd add assert of str to be string type

Reply:
added